odata-protocol/11.4 Data Modification.md

```diff
@@ -148,6 +148,21 @@ to that collection's URL. The `POST` body MUST contain a single valid
 representation of an entity of the declared target entity type,
 or one of its derived types.
 
+If the service is unable to determine the canonical collection (as defined in [section ##ContextURL])
+for the entity, it MUST fail the request.
+This determination is possible, for example, if the collection's URL
+- consists of an entity set
+- ends with a containment navigation property
+- ends with a non-containment navigation property with a navigation property binding
+
+or the payload contains a context URL.
+
+The service MUST fail the request if the
+entity represented in the payload exists already in the determined collection.
```

Comment:
Or should the POST request then be treated as an update and return 200 OK without Location header?

Reply:
No, POST to single entity URL is intentionally undefined.

Reply:
This is not about POST to a single-entity URL, but to a collection-valued non-containment navigation property, for example
```http
POST Orders(23)/Products

{"ProductID": 5}
```
if Product 5 already exists.

Reply:
So far such a request always failed, and I would not change that in a minor version.

Reply:
This seems to me like it would be contrary to REST patterns.

Reply:
2025-4-16: agreed to keep an error.